File: Medium/2785. Sort Vowels in a String.java
class Solution {
    public String sortVowels(String s) {
        List<Character> vowels = new ArrayList<>();
        for (char c : s.toCharArray()) {
            if ("aeiouAEIOU".indexOf(c) != -1) {
                vowels.add(c);
            }
        }
        Collections.sort(vowels, Collections.reverseOrder());

        StringBuilder sortedResult = new StringBuilder();
        for (char c : s.toCharArray()) {
            if ("aeiouAEIOU".indexOf(c) != -1) {
                sortedResult.append(vowels.get(vowels.size() - 1));
                vowels.remove(vowels.size() - 1);
            } else {
                sortedResult.append(c);
            }
        }
        return sortedResult.toString();
    }
}

File: Medium/2785. Sort Vowels in a String.py
VOWELS = { 'A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u' }
class Solution:
    def sortVowels(self, s: str) -> str:
        chars = list(s)
        vowels = []
        indexs = []
        for i, v in enumerate(chars):
            if v in VOWELS:
                vowels.append(v)
                indexs.append(i)
        vowels.sort()
        for i, v in zip(indexs, vowels):
            chars[i] = v
        return ''.join(chars)